### ISSUE TITLE
v4.0.0 React Native support - property atob doesn't exist

### ISSUE THREAD

**Issue by hrastnik** [auth0/jwt-decode#241]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/jwt-decode#readme) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/jwt-decode/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

React Native doesn't provide implementations for `atob` and `btoa` so the library fails when we attempt to use version 4.0.0.

### Reproduction

Install a fresh react-native project with jwt-decode and try using it.

### jwt-decode version

4.0.0

### Which browsers have you tested in?

Other

**Comment by jonkoops** [comment]
We removed the `atob()` polyfill under #162, as this is a widely supported function in browsers and Node.js. I don't think we'd want to add back a polyfill and bloat the library just for React Native. Have you tried poly-filling this function using something like [`core-js`](https://github.com/zloirock/core-js#base64-utility-methods)?

**Comment by hrastnik** [comment]
I could polyfill this function, however I think react-native has got a lot of users, and it would be nice if this library provided a way to work without it. Besides, atob is a browser API and not a JS API, and by providing a userland implementation this library would remove the browser dependency.

However, I understand your stance on the matter, and respect your decision.

**Comment by frederikprijck** [comment]
Thanks for reaching out. `jwt-decode` had always been a browser library, which happened to work on node or other platforms.

This is also called out in the repository's description:

> Decode JWT tokens; useful for browser applications.

However, as atob is supported in node, we also explicitly support node.

Any platform that has no access to atob, should rely on their own polyfill such as core-js as mentioned by @jonkoops, in order to use our library as of v4.

The reason we no longer shop an atob polyfill is because the majority has support for atob, and those that do not can use a polyfill. We believe we do not want to ship the polyfill to all our users, knowing that the majority has no need for this in the first place.

So long story short, nothing should prevent you from using this library in react-native, but you will need to bring in your own atob polyfill.

**Comment by jonkoops** [comment]
> Besides, atob is a browser API and not a JS API, and by providing a userland implementation this library would remove the browser dependency.

It might not be part of the ECMAScript specification, but it is also widely supported by most other JavaScript run-times (e.g. Node.js, Bun and Deno). However, if there are many React Native users that need it this is something we might have to reconsider adding back in.

Personally, I think it might not be worth it to add back a polyfill for this. But we should perhaps update the README with instructions specifically for React Native users to include a polyfill to help them out.

I've also logged an issue with the Hermes runtime that is used by React Native to evaluate support (see https://github.com/facebook/hermes/issues/1178). I would recommend that you chime in there and let your voice be heard.

**Comment by jonkoops** [comment]
Good news, the Hermes team has decided to implement this functionality, so polyfilling these APIs should no longer be needed in the future for React Native (see https://github.com/facebook/hermes/issues/1178#issuecomment-1813460137).

**Comment by jonkoops** [comment]
Thanks to the fast work from the Hermes team support for these function [has landed](https://github.com/facebook/hermes/issues/1178#issuecomment-1894881574), and should ship in the latest version of Hermes that will be released soon (likely 0.74).

**Comment by frederikprijck** [comment]
@iusama46 , there is no need to downgrade to v3. [Instead add the polyfill as mentioned in our readme.](https://github.com/auth0/jwt-decode?tab=readme-ov-file#polyfilling-atob)

**Comment by ShepSims** [comment]
For anyone looking for HOW to polyfill

first do
`yarn add base-64`

then at the top of your index.js add these lines
```

import { decode } from 'base-64';
global.atob = decode;
```
